src/components/ec-dropdown-search/ec-dropdown-search.vue

```diff
@@ -297,6 +347,112 @@ export default {
     },
+    onArrowKey(key) {
+      const currentItemIndex = this.filteredItems.indexOf(this.currentActiveItem);
+      let nextItem;
+
+      if (currentItemIndex >= 0) {
+        if (key === ARROW_DOWN) {
+          const selectableItems = this.filteredItems.slice(currentItemIndex + 1).filter(item => !item.disabled);
+          if (selectableItems.length) {
+            [nextItem] = selectableItems;
+          }
+        } else {
+          const selectableItems = this.filteredItems.slice(0, currentItemIndex).filter(item => !item.disabled);
+          if (selectableItems.length) {
+            nextItem = selectableItems[selectableItems.length - 1];
+          }
+        }
+      } else {
+        const selectableItems = this.filteredItems.filter(item => !item.disabled);
+        if (selectableItems.length) {
+          [nextItem] = selectableItems;
+        }
+      }
+      this.activateItemViaKeyboardNavigation(nextItem);
+    },
+    onArrowUpKeyDown() {
+      if (!this.isMultiple || this.isOpen) {
+        this.onArrowKey(ARROW_UP);
+      }
+    },
+    onArrowDownKeyDown() {
+      if (!this.isMultiple || this.isOpen) {
+        this.onArrowKey(ARROW_DOWN);
+      }
+    },
+    onEnterOrSpaceKeyDown() {
+      if (this.isOpen) {
+        if (!this.isMultiple || !this.currentActiveItem) {
+          this.closeViaKeyboardNavigation();
+        } else {
+          const keyboardNavigation = true;
+          this.select(this.currentActiveItem, keyboardNavigation);
+        }
+      } else {
+        this.show();
+      }
+    },
+    onSearchFieldEnterKeyDown() {
+      this.closeViaKeyboardNavigation();
+    },
+    onTabKeyDown() {
+      this.closeViaKeyboardNavigation();
+    },
+    activateItemViaKeyboardNavigation(item) {
+      if (item) {
+        if (this.isMultiple) {
+          this.active = item;
+          this.$refs.popover.update();
+        } else {
+          const keyboardNavigation = true;
+          this.select(item, keyboardNavigation);
+        }
+      }
+    },
+    closeViaKeyboardNavigation() {
+      if (this.isOpen) {
+        this.hide();
+        if (this.isSearchEnabled) {
+          // if the search is active the focus is lost from the trigger, then it must regain the focus
+          const $el = this.$refs.popover.$el.querySelector(this.initialFocusedElementType);
+          if ($el) {
+            $el.focus();
+          }
+          this.initialFocusedElementType = null;
+        }
+      }
+    },
+    updateScroll() {
+      const currentItemIndex = this.filteredItems.indexOf(this.currentActiveItem);
+      const $elItems = this.$refs.itemElements;
+
+      if ($elItems && $elItems.length && currentItemIndex >= 0) {
+        const itemTop = $elItems[currentItemIndex].offsetTop;
+        const itemBottom = itemTop + $elItems[currentItemIndex].offsetHeight;
+        const visibleWindowHeight = this.$refs.itemsOverflowContainer.clientHeight;
+
+        if (!this.visibleWindow) {
```

Comment:
I'm not sure that remembering the position of scrolled "window" is the right thing to do. I found multiple issues when this visibleWindow values go out of sync because I scrolled the dropdown and then I closed it without selecting a value. I think the internal state only complicates this. 

You need 4 variables:
```
let itemTopEdge = $elItems[currentItemIndex].offsetTop;
let itemBottomEdge = itemTopEdge + $elItems[currentItemIndex].offsetTop;

let containerTopEdge = this.$refs.itemsOverflowContainer.scrollTop;
let containerBottomEdge = containerTopEdge  + this.$refs.itemsOverflowContainer.clientHeight;

if (itemTopEdge < containerTopEdge) {
  this.$refs.itemsOverflowContainer.scrollTop = itemTopEdge 
} else if (itemBottomEdge  > containerBottomEdge) {
  this.$refs.itemsOverflowContainer.scrollTop = itemBottomEdge
} else {
  // do nothing, both edges of the item are visible
}
```
^^ something like this. I haven't run that code!

BTW, maybe we should also check that container's clientHeight is lower than container's scrollHeight and do the maths only if so. Otherwise there is no scrollbar needed and all items are visible.

Reply:
Yes, I like this form much better, especially for not having to save the state and can calculate it dynamically. Thank you!